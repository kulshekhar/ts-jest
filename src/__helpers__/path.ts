import { ensureDirSync } from 'fs-extra'
import { tmpdir } from 'os'
import { join, resolve } from 'path'

export const ROOT = resolve(__dirname, '..', '..')

export function tempDir(ns: string): string {
  const dir = join(tmpdir(), 'ts-jest-unit-tests', ns, parseInt((Math.random() * 1e17) as any, 10).toString(36))
  ensureDirSync(dir)

  return dir
}
