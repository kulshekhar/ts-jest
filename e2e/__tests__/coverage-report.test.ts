import * as path from 'path'

import * as fs from 'graceful-fs'
import { wrap } from 'jest-snapshot-serializer-raw'

import runJest from '../run-jest'

const DIR = path.resolve(__dirname, '../coverage-report')

test('outputs coverage report with `isolatedModules: false`', () => {
  const { stdout, exitCode } = runJest(DIR, ['--no-cache', '--coverage'], {
    stripAnsi: true,
  })
  const coverageDir = path.join(DIR, 'coverage')

  // - the `setup.js` file is ignored and should not be in the coverage report.
  // - `SumDependency.js` is mocked and the real module is never required but
  //  is listed with 0 % coverage.
  // - `notRequiredInTestSuite.js` is not required but it is listed
  //  with 0 % coverage.
  expect(wrap(stdout)).toMatchSnapshot()

  expect(() => fs.accessSync(coverageDir, fs.constants.F_OK)).not.toThrow()
  expect(exitCode).toBe(0)
})

test('outputs coverage report with `isolatedModules: true`', () => {
  const { stdout, exitCode } = runJest(DIR, ['--no-cache', '--coverage', '-c=jest-isolated.config.js'], {
    stripAnsi: true,
  })
  const coverageDir = path.join(DIR, 'coverage')

  // - the `setup.js` file is ignored and should not be in the coverage report.
  // - `SumDependency.js` is mocked and the real module is never required but
  //  is listed with 0 % coverage.
  // - `notRequiredInTestSuite.js` is not required but it is listed
  //  with 0 % coverage.
  expect(wrap(stdout)).toMatchSnapshot()

  expect(() => fs.accessSync(coverageDir, fs.constants.F_OK)).not.toThrow()
  expect(exitCode).toBe(0)
})
