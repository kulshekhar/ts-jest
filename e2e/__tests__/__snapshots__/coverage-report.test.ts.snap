// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`outputs coverage report with \`isolatedModules: false\` 1`] = `
-------------------------|---------|----------|---------|---------|-------------------
File                     | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
-------------------------|---------|----------|---------|---------|-------------------
All files                |   81.25 |        0 |   33.33 |   90.91 |                   
 file.js                 |     100 |      100 |     100 |     100 |                   
 jest-isolated.config.js |       0 |      100 |     100 |       0 | 2                 
 other-file.js           |     100 |      100 |     100 |     100 |                   
 sum-dependency.ts       |      75 |        0 |       0 |     100 | 3                 
 sum.ts                  |   88.89 |      100 |      50 |     100 |                   
-------------------------|---------|----------|---------|---------|-------------------
`;

exports[`outputs coverage report with \`isolatedModules: true\` 1`] = `
-------------------------|---------|----------|---------|---------|-------------------
File                     | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
-------------------------|---------|----------|---------|---------|-------------------
All files                |   81.25 |        0 |   33.33 |   90.91 |                   
 file.js                 |     100 |      100 |     100 |     100 |                   
 jest-isolated.config.js |       0 |      100 |     100 |       0 | 2                 
 other-file.js           |     100 |      100 |     100 |     100 |                   
 sum-dependency.ts       |      75 |        0 |       0 |     100 | 3                 
 sum.ts                  |   88.89 |      100 |      50 |     100 |                   
-------------------------|---------|----------|---------|---------|-------------------
`;
